docs/content/latest/api/ysql/datatypes/type_array/literals/text-typecasting-and-literals.md

```diff
@@ -0,0 +1,242 @@
+
+## Stating the general rule
+
+Here is the general rule.
+
+- The literal for a value of any data type is the possibly enquoted bare text that specifies the intended value followed by the typecast operator to the desired target data type.
+- This rule is applied recursively, for the literal for a composite value, but with different actual rules at different levels of nesting. For example, the literal for an array value as a whole must be typecasted. But, because the data type of every value in the array is already determined, the bare text that specifies these values is _not_ typecasted.
+- The `::text` typecast of any value can always be used as the bare text of the literal that will recreate that value.
+
+A simple way to see examples of the bare text that specifies an array value, when used as a component in the literal, is to create the value using the constructor and then to inspect its `::text` typecast. But, of course, the safe way to create the text of a literal for an intended value is to understand the syntax and semantics that govern its composition.
+
+When we need to make this distinction, throughout the whole of this major _"[Arrays](../../../type_array/)"_ section, we shall distinguish between:
+
+- (1) the _literal_ for the intended value (i.e. the whole thing with the enquoting and the typecast, when one or both of these are needed);
+- and (2) the _text of the literal_ for the intended value (i.e. the bare text that specifies this value).
+
+**Note:** often, the data type of the assignment target (for example, (a field in a column in a schema-level table or a variable in a PL/pgSQL program) is sufficient implicitly to specify the intended typecast without writing the operator. But it's never harmful to write it. Moreover, in some cases, omitting the explicit typecast can bring performance costs. For this reason, Yugabyte recommends that you always write the explicit typecast unless you are certain beyond all doubt that omitting it brings no penalty.
+
+## Defining the "canonical form of a literal"
+
+The term _"canonical form"_ applies specifically to the _text of a literal_ rather than to the _literal as a whole_. But when the text of a literal is in canonical form, we say that the literal as a whole, too, is in canonical form.
+
+The canonical form of the text of a literal that produces a specific value, of any data type, is quite simply the `::text` typecast of that value.
+
+We shall see in the remaining sections in this enclosing section that many spellings of the text of an array literal, in addition to the canonical form, will produce a particular intended target value. The differences are due to how whitespace, punctuation, and escape characters are used.
```

Comment:
```suggestion
In the remaining sections in this enclosing section, you will see that many different spellings of the text of an array literal, in addition to the canonical form, will produce a particular intended target value. The differences are due to how whitespace, punctuation, and escape characters are used.
```

Reply:
Strange. The sentence that you want to change is still in place in what I pushed. But this is marked "Outdated". I made a not do make the change manually when the leading copy of my corpus is back in place on my laptop.